geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AbstractAvroSimpleFeatureTest.scala

```diff
@@ -0,0 +1,93 @@
+import java.util.UUID
+
+import com.vividsolutions.jts.geom.{Point, Polygon}
+import org.geotools.factory.Hints
+import org.geotools.filter.identity.FeatureIdImpl
+import org.locationtech.geomesa.features.avro.serde.Version2ASF
+import org.locationtech.geomesa.utils.geohash.GeohashUtils
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.text.WKTUtils
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Random
+
+trait AbstractAvroSimpleFeatureTest {
+
+  val complexSchema = "f0:String,f1:Integer,f2:Double,f3:Float,f4:Boolean,f5:UUID,f6:Date,f7:Point:srid=4326,"+
+    "f8:Polygon:srid=4326,f9:Long,f10:String,f11:Integer,f12:Date,f13:Geometry,f14:UUID"
+  val complexSft = SimpleFeatureTypes.createType("test", complexSchema)
+
+  def createComplicatedFeatures(numFeatures : Int) : List[Version2ASF] = {
+    val r = new Random()
+    r.setSeed(0)
+
+    val list = new ListBuffer[Version2ASF]
+    for(i <- 0 until numFeatures){
+      val fid = new FeatureIdImpl(r.nextString(5))
+      val sf = new Version2ASF(fid, complexSft)
+
+      sf.setAttribute("f0", r.nextString(10).asInstanceOf[Object])
+      sf.setAttribute("f1", r.nextInt().asInstanceOf[Object])
+      sf.setAttribute("f2", r.nextDouble().asInstanceOf[Object])
+      sf.setAttribute("f3", r.nextFloat().asInstanceOf[Object])
+      sf.setAttribute("f4", r.nextBoolean().asInstanceOf[Object])
+      sf.setAttribute("f5", UUID.fromString("12345678-1234-1234-1234-123456789012"))
+      sf.setAttribute("f6", new SimpleDateFormat("yyyyMMdd").parse("20140102"))
+      sf.setAttribute("f7", GeohashUtils.wkt2geom("POINT(45.0 49.0)").asInstanceOf[Point])
+      sf.setAttribute("f8", GeohashUtils.wkt2geom("POLYGON((-80 30,-80 23,-70 30,-70 40,-80 40,-80 30))").asInstanceOf[Polygon])
+      sf.setAttribute("f9", r.nextLong().asInstanceOf[Object])
+
+      // test nulls on a few data types
+      "f10,f11,f12,f13,f14".split(",").foreach { id =>
+        sf.setAttribute(id, null.asInstanceOf[Object])
+      }
+
+      list += sf
+    }
+    list.toList
+  }
+
+  val simpleSft = SimpleFeatureTypes.createType("AvroSimpleFeatureWriterTest", "name:String,*geom:Point,dtg:Date")
+
+  def createSimpleFeature: SimpleFeature = {
+    val builder = AvroSimpleFeatureFactory.featureBuilder(simpleSft)
+    builder.reset()
+    builder.set("name", "test_feature")
+    builder.set("geom", WKTUtils.read("POINT(-110 30)"))
+    builder.set("dtg", "2012-01-02T05:06:07.000Z")
+
+    val sf = builder.buildFeature("fid")
+    sf.getUserData.put(Hints.USE_PROVIDED_FID, java.lang.Boolean.TRUE)
+    sf
+  }
+
+  def getFeatures(f: File) = {
+    val dfr = new AvroDataFileReader(new FileInputStream(f))
+    try {
+      dfr.toList
+    } finally {
+      dfr.close()
+    }
+  }
+
+  def getTmpFile = {
+    val tmpFile = File.createTempFile("geomesa", "avro")
+    tmpFile.deleteOnExit()
```

Comment:
[picky] deleteOnExit only deletes if the jvm shuts down safely (ie the process is not killed), maybe this is fine.